pkg/scalers/loki_scaler.go

```diff
@@ -0,0 +1,291 @@
+
+	return &lokiScaler{
+		metricType: metricType,
+		metadata:   meta,
+		httpClient: httpClient,
+		logger:     logger,
+	}, nil
+}
+
+func parseLokiMetadata(config *ScalerConfig) (meta *lokiMetadata, err error) {
+
+	meta = &lokiMetadata{}
+
+	if val, ok := config.TriggerMetadata[lokiServerAddress]; ok && val != "" {
+		meta.serverAddress = val
+	} else {
+		return nil, fmt.Errorf("no %s given", lokiServerAddress)
+	}
+
+	if val, ok := config.TriggerMetadata[lokiQuery]; ok && val != "" {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no %s given", lokiQuery)
+	}
+
+	if val, ok := config.TriggerMetadata[lokiThreshold]; ok && val != "" {
+		t, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", lokiThreshold, err)
+		}
+
+		meta.threshold = t
+	} else {
+		return nil, fmt.Errorf("no %s given", lokiThreshold)
+	}
+
+	meta.activationThreshold = 0
+	if val, ok := config.TriggerMetadata[lokiActivationThreshold]; ok {
+		t, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, fmt.Errorf("activationThreshold parsing error %s", err.Error())
+		}
+
+		meta.activationThreshold = t
+	}
+
+	if val, ok := config.TriggerMetadata[tenantName]; ok && val != "" {
+		meta.tenantName = val
+	}
+
+	meta.ignoreNullValues = lokiDefaultIgnoreNullValues
+	if val, ok := config.TriggerMetadata[lokiIgnoreNullValues]; ok && val != "" {
+		ignoreNullValues, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("err incorrect value for ignoreNullValues given: %s, "+
+				"please use true or false", val)
+		}
+		meta.ignoreNullValues = ignoreNullValues
+	}
+
+	meta.unsafeSsl = false
+	if val, ok := config.TriggerMetadata[unsafeSsl]; ok && val != "" {
+		unsafeSslValue, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %s: %s", unsafeSsl, err)
+		}
+
+		meta.unsafeSsl = unsafeSslValue
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+
+	// parse auth configs from ScalerConfig
+	meta.lokiAuth, err = authentication.GetAuthConfigs(config.TriggerMetadata, config.AuthParams)
+	if err != nil {
+		return nil, err
+	}
+
+	return meta, nil
+}
+
+// IsActive validates the loki query result against the activation threshold
+func (s *lokiScaler) IsActive(ctx context.Context) (bool, error) {
+	val, err := s.ExecuteLokiQuery(ctx)
+	if err != nil {
+		s.logger.Error(err, "error executing loki query")
+		return false, err
+	}
+
+	return val > s.metadata.activationThreshold, nil
+}
+
+// Close returns a nil error
+func (s *lokiScaler) Close(context.Context) error {
+	return nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *lokiScaler) GetMetricSpecForScaling(context.Context) []v2.MetricSpec {
+	metricName := kedautil.NormalizeString("loki")
+	externalMetric := &v2.ExternalMetricSource{
+		Metric: v2.MetricIdentifier{
+			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, metricName),
```

Comment:
I'd put "loki" here directly, removing the line 181 because it isn't needed
```suggestion
			Name: GenerateMetricNameWithIndex(s.metadata.scalerIndex, "loki"),
```
I mean, "loki" is already normalized so we don't need to do it